src/lib.rs

```diff
@@ -601,7 +601,27 @@ impl<'a> WrapIterImpl<'a> {
         while let Some((idx, ch)) = self.char_indices.next() {
             let char_width = ch.width().unwrap_or(0);
             let char_len = ch.len_utf8();
-            if is_whitespace(ch) {
+
+            if ch == '\n' {
+                self.split = idx;
+                self.split_len = char_len;
+                self.line_width_at_split = self.line_width;
+                self.in_whitespace = false;
+
+                // If this is not the final line, return the current line. Otherwise,
+                // we will return the line with its line break after exiting the loop
+                if self.split + self.split_len < self.source.len() {
+                    let mut line = self.create_result_line(wrapper);
+                    cow_add_assign(&mut line, &self.source[self.start..self.split]);
+
+                    self.start = self.split + self.split_len;
+                    self.line_width += wrapper.subsequent_indent.width();
+                    self.line_width -= self.line_width_at_split;
```

Comment:
I believe this is redundant -- in total, `self.line_width` becomes `wrapper.subsequent_indent.width()` since you set it to `self.line_width_at_split` above. So you can remove this line and simplify the previous line to a direct assignment: 
```
self.line_width = wrapper.subsequent_indent.width();
```